kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/TopicEncryptionST.java

```diff
@@ -0,0 +1,127 @@
+import io.kroxylicious.systemtests.installation.vault.Vault;
+import io.kroxylicious.systemtests.resources.vault.VaultTestKekManager;
+import io.kroxylicious.systemtests.steps.KafkaSteps;
+import io.kroxylicious.systemtests.steps.KroxyliciousSteps;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(KroxyliciousExtension.class)
+class TopicEncryptionST extends AbstractST {
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final Logger LOGGER = LoggerFactory.getLogger(TopicEncryptionST.class);
+    private static Vault vaultOperator;
+    private static Kroxylicious kroxylicious;
+    private final String clusterName = "my-cluster";
+
+    @BeforeAll
+    void setUp() {
+        vaultOperator = new Vault(Constants.VAULT_DEFAULT_NAMESPACE);
+        NamespaceUtils.createNamespaceWithWait(Constants.VAULT_DEFAULT_NAMESPACE);
+        vaultOperator.deploy();
+
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KROXY_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.warn("Skipping kafka deployment. It is already deployed!");
+            return;
+        }
+        LOGGER.info("Deploying Kafka in {} namespace", Constants.KROXY_DEFAULT_NAMESPACE);
+
+        Kafka kafka = KafkaTemplates.kafkaPersistentWithKRaftAnnotations(Constants.KROXY_DEFAULT_NAMESPACE, clusterName, 3).build();
+
+        resourceManager.createResourceWithWait(
+                KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka, 3).build(),
+                kafka);
+    }
+
+    @AfterAll
+    void cleanUp() throws IOException {
+        if (vaultOperator != null) {
+            vaultOperator.delete();
+        }
+    }
+
+    @Test
+    void produceAndConsumeEncryptedMessage(String namespace) {
+        String topicName = "my-topic";
+        String message = "Hello-world";
+        int numberOfMessages = 1;
+        String consumedMessage = message + " - " + (numberOfMessages - 1);
+
+        var testKekManager = new VaultTestKekManager(Constants.VAULT_DEFAULT_NAMESPACE, Constants.VAULT_SERVICE_NAME + "-0");
+        testKekManager.generateKek(topicName);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithTopicEncryptionFilter(clusterName, 1, topicName);
+        String bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And KafkaTopic in {} namespace", namespace);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
```

Comment:
I feel like this should be in a `@BeforeEach` method

Reply:
From my POV we should keep this inside the test because we need to run each test using a different topic to avoid clashing and/or rubbish

Reply:
But that is exactly what a beforeEach will do, note it's not the same as the current beforeAll

Reply:
Right, but we can do it only if the replicas for kroxylicious are the same for all tests. Anyway I'll do something